### DISCUSSION TITLE
Translate text on client side

### DISCUSSION THREAD

**Discussion by romawork99** [argosopentech/argos-translate#145] [Q&A]
I want to know if it's possible to run argos-translate on client side(web browser) using javascript instead of running on server with python? 

P.S.
Thanks in advance and sorry if it's a stupid question, I'm new to this.

**Comment by pierotofy** [comment]
Purely on the client-side, no. You might need something like https://github.com/LibreTranslate/LibreTranslate to access the API via Javascript (but this still requires a server).

**Reply by romawork99** [reply]
thanks for the answer, I'm considering using LibreTranslate, maybe it's possible to convert the model to tensorflow.js and run on client side?

**Reply by pierotofy** [reply]
Maybe!

**Comment by PJ-Finlay** [comment]
It might be possible but would be difficult. You would need to run Python, OpenNMT, and SentencePiece from WebAssembly. If you need client side translations it's probably easiest to use the PyQt GUI, and if you need to run in a web browser LibreTranslate.

**Reply by romawork99** [reply]
I want to run it completely in browser, using laptop resources, without back-end server with python, for example, like chrome extension or like static web page, when all javascript files downloaded on client side and not making any requests to back end for translation
